packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -58,7 +67,7 @@ export default class RequestClient {
   }
 
   headers () {
-    const userHeaders = this.opts.companionHeaders || {}
+    const userHeaders = this.#companionHeaders || {}
     return Promise.resolve({
       ...RequestClient.defaultHeaders,
       ...userHeaders,
```

Comment:
We don't need the `|| {}`, the spread operator already does the smart thing (it's treats non-plain-object as empty objects).
```suggestion
      ...this.#companionHeaders,
```